test/files-helper.js

```diff
@@ -2,90 +2,132 @@
 
 var fs = require('fs');
 var path = require('path');
+var temp = require('temp');
+var scriptName = require('../package.json').name;
 
 module.exports = FilesHelper;
 
-function FilesHelper(config, done) {
-  this.testRepoDir = path.resolve(__dirname, 'site_builder_test');
-  this.gemfile = path.resolve(this.testRepoDir, 'Gemfile');
-  this.pagesConfig = path.resolve(this.testRepoDir, config.pagesConfig);
-  this.configYml = path.resolve(this.testRepoDir, '_config.yml');
-  this.internalConfig = path.resolve(this.testRepoDir, '_config_internal.yml');
-  this.externalConfig = path.resolve(this.testRepoDir, '_config_external.yml');
-  this.filesToDelete = [];
-  this.lockDir = path.resolve(__dirname, 'site_builder_test_lock_dir');
-  this.lockfilePath = path.resolve(this.lockDir, '.update-lock-repo_name');
-  fs.mkdir(this.lockDir, '0700', done);
+function FilesHelper() {
 }
 
-FilesHelper.prototype.afterEach = function(done) {
-  var that = this;
-  var removePromise = this.removeFile(this.configYml);
+FilesHelper.prototype.init = function(config) {
+  var helper = this;
 
-  this.filesToDelete.map(function(fileToDelete) {
-    removePromise = removePromise
-      .then(function() { return that.removeFile(fileToDelete); });
+  return new Promise(function(resolve, reject) {
+    temp.mkdir(scriptName + '-test-files-', function(err, tempDir) {
+      var testRepoDir = path.resolve(tempDir, 'site_builder_test'),
+          lockDir = path.resolve(tempDir, 'site_builder_test_lock_dir');
+
+      if (err) {
+        return reject(err);
+      }
+
+      fs.mkdir(lockDir, '0700', function(err) {
+        if (err) {
+          return reject(err);
+        }
+        initHelper(helper, config, tempDir, testRepoDir, lockDir);
+        resolve();
+      });
+    });
   });
-  this.filesToDelete = [];
+};
+
+function initHelper(helper, config, tempDir, testRepoDir, lockDir) {
+  helper.dirs = {
+    testRepoDir: testRepoDir,
+    lockDir: lockDir,
+    tempDir: tempDir
+  };
+
+  helper.files = {
+    gemfile: path.resolve(testRepoDir, 'Gemfile'),
+    pagesConfig: path.resolve(testRepoDir, config.pagesConfig),
+    configYml: path.resolve(testRepoDir, '_config.yml'),
+    internalConfig: path.resolve(testRepoDir, '_config_internal.yml'),
+    externalConfig: path.resolve(testRepoDir, '_config_external.yml'),
+    lockfilePath: path.resolve(lockDir, '.update-lock-repo_name')
+  };
+
+  helper.filesToDelete = [];
+}
+
+FilesHelper.prototype.afterEach = function() {
+  var helper = this,
+      files = helper.filesToDelete.slice();
 
-  removePromise = removePromise
-    .then(function() { return that.removeRepoDir(); })
-    .then(done, done);
+  helper.filesToDelete = [];
+  files = files.concat(Object.keys(helper.files).map(function(key) {
+    return helper.files[key];
+  }));
+
+  return removeItems(files, 'unlink')
+    .then(function() {
```

Comment:
What happens when `removeItems` throws an error here? Does it need to be taken care of in some way?